src/jsonapi-datastore.js

```diff
@@ -11,10 +11,68 @@
     this.id = id;
     this._type = type;
     this._attributes = [];
+    this._dependents = [];
     this._relationships = [];
   }
 
   /**
+   * Add a dependent to a model.
+   * @method _addDependence
+   * @param {string} type The type of the dependent model.
+   * @param {string} id The id of the dependent model.
+   * @param {string} key The name of the relation found on the dependent model.
+    */
+  _addDependence(type, id, key) {
+    var self = this,
+      found;
+
+    found = self._dependents.find(function(dependent) {
+      return dependent.id === id && dependent.type === type && dependent.relation === key;
+    });
+    if (found === undefined) {
+      self._dependents.push({id: id, type: type, relation: key});
+    }
+  }
+
+  /**
+   * Removes a dependent from a model.
+   * @method _removeDependence
+   * @param {string} type The type of the dependent model.
+   * @param {string} id The id of the dependent model.
+    */
+  _removeDependence(type, id) {
+    var self = this,
+      found;
+
+    self._dependents.forEach(function(val, idx) {
```

Comment:
Here we could use `find` since we want to guarantee that those records are unique (provided we add a `key` parameter to the `_removeDependence` method). It would be slightly longer, but I believe would make the purpose of the code clearer.